cylc/uiserver/handlers.py

```diff
@@ -386,6 +390,18 @@ def open(self, *args, **kwargs):  # noqa: A003
         )
 
     async def on_message(self, message):
+        with suppress(ValueError, SyntaxError):
+            message_dict = ast.literal_eval(message)
+            op_name = ''
+            op_id = message_dict.get("id", None)
+            with suppress(KeyError):
+                op_name = message_dict['payload']['operationName']
+            if (message_dict['type'] == 'start'
+                and op_name
```

Comment:
Do we need the `op_name`?

Could this be:

```python
self.sub_statuses[op_id] = message_dict.get('type')
```

(no need to check if the operation name is `LogData`?)

Reply:
Since the only subscriptions we need to track in this dictionary are log data ones, and other subscriptions come in with a start (deltas for example) we need to check for the operation name to ensure we don't add to the dictionary anything that isn't a log data subscription. I think I will rename `sub_statuses`  to `log_sub_statuses` to make this clearer.
Note that if you start a Log subscription in graphiql,depending on how you write the subscription, it may not come in with the `LogData` operation name. In this case we add the op_id to the `sub_statuses` dictionary in `resolvers.py` code to ensure it doesn't get missed.

Reply:
I think it's ok to make this generic rather than targetting it at the LogData operation manually (so long as we remove operations from the dict when appropriate).

(note that subscriptions which aren't called `LogData` will fail)